nucypher/cli/stake.py

```diff
@@ -0,0 +1,309 @@
+        else:
+            # TODO: Validate the balance of self-manged funders.
+            fund_now = False
+
+        # Validate balance
+        if STAKEHOLDER.funding_tokens == 0 and fund_now:
+            click.secho(f"{STAKEHOLDER.funding_account} has 0 NU.")
+            raise click.Abort
+        if not quiet:
+            click.echo(f"Funding account balance | {STAKEHOLDER.funding_tokens} | {STAKEHOLDER.funding_eth} ETH")
+
+
+        #
+        # Stage Stake
+        #
+
+        if not value:
+            min_locked = STAKEHOLDER.economics.minimum_allowed_locked
+            value = click.prompt(f"Enter stake value in NuNits", type=STAKE_VALUE, default=min_locked)
+        value = NU.from_nunits(int(value))
+
+        if not duration:
+            prompt = f"Enter stake duration ({STAKEHOLDER.economics.minimum_locked_periods} period min)"
+            duration = click.prompt(prompt, type=STAKE_DURATION)
+
+        start_period = STAKEHOLDER.staking_agent.get_current_period()
+        end_period = start_period + duration
+
+        #
+        # Review
+        #
+
+        if not force:
+            painting.paint_staged_stake(ursula=STAKEHOLDER,
+                                        stake_value=value,
+                                        duration=duration,
+                                        start_period=start_period,
+                                        end_period=end_period)
+
+            confirm_staged_stake(stakeholder=STAKEHOLDER, value=value, duration=duration)
+
+        # Last chance to bail
+        target = staking_address or "derived account"
+        message = f"Transfer {value.to_nunits()} NuNits and {STAKEHOLDER.eth_funding} wei to {target}?"
+        if not force:
+            if fund_now:
+                click.confirm(message, abort=True)
+            click.confirm("Publish staged stake to the blockchain?", abort=True)
+
+        # Execute
+        new_stake = STAKEHOLDER.initialize_stake(amount=value,
+                                                 duration=duration,
+                                                 checksum_address=staking_address,
+                                                 password=password,
+                                                 fund_now=fund_now)
+
+        painting.paint_staking_confirmation(ursula=STAKEHOLDER, transactions=new_stake.transactions)
+        return  # Exit
+
+    elif action == 'divide':
+        """Divide an existing stake by specifying the new target value and end period"""
+
+        if staking_address and index is not None:
+            staker = STAKEHOLDER.get_active_staker(address=staking_address)
+            current_stake = staker.stakes[index]
+        else:
+            current_stake = select_stake(stakeholder=STAKEHOLDER)
+
+        #
+        # Stage Stake
+        #
+
+        # Value
+        if not value:
+            value = click.prompt(f"Enter target value (must be less than {str(current_stake.value)})", type=STAKE_VALUE)
```

Comment:
Should the prompt message be `less than or equal to`?

Reply:
Also, does `type=STAKE_VALUE` ensure that the value entered is indeed less than or equal to `current_stake.value`? If not should we check it here to fail sooner.

Reply:
Looks like its inclusive on both ends documented here: <https://click.palletsprojects.com/en/7.x/options/#range-options>